neuro_san/internals/chat/chat_history_message_processor.py

```diff
@@ -25,10 +26,24 @@ class ChatHistoryMessageProcessor(MessageProcessor):
     in a chat history.
     """
 
-    def __init__(self):
+    def __init__(self, max_message_history: int = None):
         """
         Constructor
+
+        :param max_message_history: The maximum number of messages to preserve
+                in the message history, not including the instructions.
+                The default value of None implies there is no maximum.
+                Non-positive numbers revert to no-maximum behavior.
         """
+        self.max_message_history: int = max_message_history
+        if self.max_message_history is not None:
+            if not isinstance(self.max_message_history, Number):
+                # If we don't have a number we don't have a max.
+                self.max_message_history = None
+            else:
+                # Be sure we are dealing with an integer
+                self.max_message_history = int(self.max_message_history)
```

Comment:
Be sure we get either None or an integer for the new max_message_history parameter.

Reply:
And said integer is positive?

Reply:
That is checked below.